### DISCUSSION TITLE
AWS S3 Signed URL PNG Files getting corrupted post uploading on S3

### DISCUSSION THREAD

**Discussion by Abhman229** [aws/aws-sdk-java#2803] [Q&A]
I'm using JAVA AWS SDK 1.12.17 to obtained signed URLs where I'm uploading the files (PNG/JPEG/PDF), JPEG/PDF Files are working fine however with PNG files are getting corrupted when I'm downloading them manually from s3 or in code. Please suggest.

**Comment by debora-ito** [comment]
Can you share a code sample we can use to reproduce the issue?

**Reply by Abhman229** [reply]
I'm using aws sdk and there's AmazonS3Client.class that has one method as getObject. So in my code I'm fetching this object from S3 and writing it to a file and then appending that file with the correct extension (I'm saving the file extension in db). But when I'm trying to view this its coming up as an corrupted file. Also for generating the signed url I'm using AmazonS3Client.class generatePresignedUrl method . 
Same code works fine in case of JPEG/JPG/PDF but only corrupts in PNG.

Please suggest, hope to hear from you soon

**Reply by Abhman229** [reply]
Hi @debora-ito , any update on this ?

**Reply by debora-ito** [reply]
Can you share a minimal reproducible code I can simply copy+paste and run into the same error?

I understand by your description, but having a repro code makes the investigation way faster - See http://sscce.org/
